src/lib/modes/mode_pad/mode_pad.h

```diff
@@ -63,16 +76,36 @@ class BOTAN_TEST_API BlockCipherModePaddingMethod {
       * virtual destructor
       */
       virtual ~BlockCipherModePaddingMethod() = default;
+
+   protected:
+      /**
+      * Applies the concrete padding to the @p last_block assuming the padding
+      * bytes should start at @p padding_start_pos within the last block.
+      *
+      * Concrete implementations of this function must ensure not to leak
+      * @p padding_start_pos via side channels.
```

Comment:
This should note that this function is always called with the input span in a poisoned state